### ISSUE TITLE
Browser runs cached JS files even when new version are pushed

### ISSUE THREAD

**Issue by jspenger** [portals-project/playground#12]
The browser will not re-download previously downloaded JS files if they are in some cache. For this reason, even though we have updated the runtime to a new version, the old version still runs in my browser. We should have some mechanism for invalidating the cached downloads on the client side. Although this is a deeper issue, perhaps the following link gives a small taste of it https://stackoverflow.com/questions/32414/how-can-i-force-clients-to-refresh-javascript-files.

**Comment by MKLepium** [comment]
Locally using Jekyll:

We could use a '_config.yml' file and use a config like this:


```
defaults:
  -
    scope:
      path: ""
    values:
      cache-control: "no-cache, no-store, must-revalidate"
```

This forces Jekyll to always send a header similar to this

```
HTTP/1.1 200 OK
Etag: ca453-bac-6480f11d
Content-Type: text/html; charset=utf-8
Content-Length: 2988
Last-Modified: Wed, 07 Jun 2023 21:05:33 GMT
Cache-Control: private, max-age=0, proxy-revalidate, no-store, no-cache, must-revalidate <-
Server: WEBrick/1.6.0 (Ruby/2.7.0/2019-12-25)
Date: Wed, 07 Jun 2023 21:12:02 GMT
Connection: Keep-Alive
```
Including the necessary changes.

As for the usage with github pages:

The header of Github pages is and will always be:

`cache-control: max-age=600`

Github pages does not allow for a Cache-control "change". The headers sent by the web-server are always determined by them and they have confirmed that there is no option of changing them 
https://webapps.stackexchange.com/questions/119286/caching-assets-in-website-served-from-github-pages


One option that can be considered is using Github Actions to automatically insert a version number to the file and change the according line in the html. However this seems unreasonably worked up for a problem like this.

A more "hacky" (but working),  solution is something like this:
```
      <script>
        function appendScriptWithTimestamp(scriptSrc) {
          var scriptElement = document.createElement('script');
          scriptElement.src = scriptSrc + '?v=' + Date.now();
          document.head.appendChild(scriptElement);
        }
        // Example Programs
        appendScriptWithTimestamp('assets-playground/js/examples.js');
        // Setup Codemirror and Example Swaps
        appendScriptWithTimestamp('assets-playground/js/jsEditorRunner.js');
      </script>
```

With this approach, each time the page is loaded, the script elements will not be initially attached within the HTML. However, upon loading, they will be dynamically added to the <head> section of the document using JavaScript. The appendScriptWithTimestamp function is responsible for creating the script elements and setting the appropriate source URLs with a unique timestamp to ensure they are not cached.

While this solution definetly is a hack, it effectively ensures that the latest versions of the JavaScript files are always loaded without relying on caching mechanisms.

#15 shows a working implementation of this.

**Comment by jspenger** [comment]
It seems as though it is the more common option to tag the file names, either through adding an explicit version tag like `?v=...`, or through explicitly renaming the files. I think in the interest of keeping the solution as simple as possible for now, I would suggest that we manually tag the imported files for now with `?v=1`, etc. monotonically increasing between commits. This is for two reasons: 1) it is the least complex solution, anyone should immediately be able to grasp how it works and what to do; and 2) we can manually trigger refreshing the files when we want to roll-out larger changes. 

So, do the changes according to the comments here, and then we can close this issue for now.

**Comment by MKLepium** [comment]
Changes according to your request are now here #15.

**Comment by jspenger** [comment]
closed by #15
